File: leetcode-blind75/IsValid.cs
public class Solution {
    public bool IsValid(string s) {
        int len = s.Length;
        Stack <char> parenStack = new();
        
        foreach(char item in s)
        {
            switch(item) 
            {
                case '{':
                    parenStack.Push('}');
                    break;
                case '[':
                    parenStack.Push(']');
                    break;
                case '(':
                    parenStack.Push(')');
                    break;
                case '}':
                case ']':
                case ')':
                    if (parenStack.Count == 0)
                    {
                        return false;
                    }
                
                    if (parenStack.Pop() != item)
                    {
                        return false;
                    }
                    
                    break;
                
                default:
                break;
            }
            
        }
        return parenStack.Count == 0;
    }
}

File: NeetCode_150/Stack/isValid.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
if (s.length == 1) {
    return false;
}  
    
let hashMap = {")": "(", "}": "{", "]": "["};
let stack = [];

for (let i = 0; i < s.length; i++) {
    if (hashMap.hasOwnProperty(s[i])) {
        if (stack.length != 0 && stack[stack.length-1] == hashMap[s[i]]) {
            stack.pop();
        } else {
            return false;
        }
    } else {
        stack.push(s[i]);
    }
}
    return stack.length == 0 ? true : false;
};